codegen/src/main/java/software/amazon/awssdk/codegen/docs/AsyncOperationDocProvider.java

```diff
@@ -150,4 +154,33 @@ protected void applyParams(DocumentationBuilder docBuilder) {
                              opModel.getInputShape().getC2jName());
         }
     }
+
+    /**
+     * Provider for traditional paginated method that takes in a request object and returns a response object.
+     */
+    private static class AsyncPaginated extends AsyncOperationDocProvider {
+
+        private AsyncPaginated(IntermediateModel model, OperationModel opModel) {
+            super(model, opModel);
+        }
+
+        @Override
+        protected String appendToDescription() {
+            return opModel.isPaginated() ? paginationDocs.getDocsForAsyncOperation() : null;
```

Comment:
return null will cause a `Null Dereference` risk on fortify, we can probably return Optional here